### PR TITLE
Added 2FA for app-generated tokens

### PR DESCRIPTION
**Pull request by enjikaka**
This fix let users with 2FA log in if they generate their tokens with an app.

### CODE REVIEW

**Comment by anthonny** [comment]
:+1: 

Thank you @enjikaka !
I'll merge this PR in the raw source asap ;)

File: hubpress/scripts/vendor.js [hunk 1]
```diff
@@ -69,12 +68,19 @@
           if (this.readyState == 4) {
             if (this.status >= 200 && this.status < 300 || this.status === 304) {
               cb(null, raw ? this.responseText : this.responseText ? JSON.parse(this.responseText) : true, this);
+            } else if (this.status === 401) {
+            	var tfc = prompt("Enter your temporary two-factort authentication code from your app.");
```
Comment (cmfcmf) [inline hubpress/scripts/vendor.js]: Typo: *two-factor*

Comment (enjikaka) [inline hubpress/scripts/vendor.js]: I was going to fix this but forgot before creating a pull request. Thanks for the reminder! :+1:

File: hubpress/scripts/vendor.js [hunk 2]
```diff
@@ -69,12 +68,19 @@
...
+            		//this = Github(options);
+            		_request(method, path, data, cb, raw, sync);
+            	}
             } else {
               cb({path: path, request: this, error: this.status});
             }
           }
         };
-        }
+       }
```
Comment (jgillman) [inline hubpress/scripts/vendor.js]: Looks like this should be either one space to the left or one to the right.

**Comment by jgillman** [comment]
@enjikaka Looks like there's some minor formatting issues on lines 96, 100, 102, and 105. Line 96 has a tab which should be two spaces and the other lines have only one space of indentation instead of two.

Formate nick-picking aside, thank you for adding in 2FA! Your fix is very succinct.

**Comment by enjikaka** [comment]
I'll look over the whitespace.
Den 10 feb 2015 19:58 skrev "Joel Gillman" <notifications@github.com>:

**Comment by anthonny** [comment]
@enjikaka i'll merge your code, but a part will disappear as i said on the other project.
Thank you again

**Comment by anthonny** [comment]
Done! i hope you will like the UX

**Comment by enjikaka** [comment]
Nice! That was somewhat similar to my first pull request, haha: https://github.com/HubPress/hubpress.io/pull/30

Thanks for implementing! :) Partly related; if you haven't put in username and repo-name in the config.json, it'd be nice if the log in screen told you, i.e. if the default values has not been changed to your personal stuff.

**Comment by anthonny** [comment]
@enjikaka :beers: 
I agree for the default values => #46

**Comment by enjikaka** [comment]
@anthonny Thanks! :smile:

**Pull request merged**